### PR TITLE
Implement create fingerprint route

### PR DESCRIPTION
**Pull request by andela-kakpobome**
### What does this PR do?

It fleshes out the endpoint that receives an audio file(full length, not the sample). It validates the uploaded file and returns the generated fingerprint.

### CODE REVIEW

File: api/node-echoprint-server/controllers/api.js [hunk 1]
```diff
@@ -66,7 +81,7 @@ exports.ingest = function(req, res) {
   if (!codeVer)
     res.status(500).send({ error: 'Missing "version" field' });
   if (codeVer != config.codever)
-    res.status(500).send({ error: 'Version "' + codeVer + '" does not match required version "' + config.codever + '"' }););
+    res.status(500).send({ error: 'Version "' + codeVer + '" does not match required version "' + config.codever + '"' });
```
Comment (makinwab) [inline api/node-echoprint-server/controllers/api.js]: Use string interpolation.

Comment (makinwab) [reply api/node-echoprint-server/controllers/api.js]: https://github.com/airbnb/javascript

File: api/node-echoprint-server/controllers/api.js [hunk 2]
```diff
@@ -100,3 +115,73 @@ exports.ingest = function(req, res) {
...
+
+/**
+* Upload
+*
+* Validates and uploads files
+*
+* @param {Object} req
+* @param {Object} res
+*/
+function upload(req,res) {
```
Comment (makinwab) [inline api/node-echoprint-server/controllers/api.js]: ⚠️ I believe that I will never merge until this is sorted.

You have context. Willingness to remember.

File: api/node-echoprint-server/controllers/api.js [hunk 3]
```diff
@@ -100,3 +115,73 @@ exports.ingest = function(req, res) {
...
+
+  upload(req,res)
+  .then((file) => {
+
+    //generate the Fingerprint
+    echoprintCodegen(file.path,(err, data) => {
+      if (err) res.status(500).send({error:err});
+
+      //we're done, clear all uploads
+      glob('./uploads/**',(err,files) => {
```
Comment (makinwab) [inline api/node-echoprint-server/controllers/api.js]: We can have this in another method. `#SRP`

**Comment by makinwab** [comment]
Reminder to add ESlint.

File: api/node-echoprint-server/controllers/api.js (hunk 1 above)
Comment (andela-kakpobome) [reply api/node-echoprint-server/controllers/api.js]: Yes ma'am @andela-bmakinwa

File: api/node-echoprint-server/controllers/api.js (hunk 2 above)
Comment (andela-kakpobome) [reply api/node-echoprint-server/controllers/api.js]: The thunder that you will experience, would be brought to you by 2 angry people!

Comment (makinwab) [reply api/node-echoprint-server/controllers/api.js]: Never

**Pull request merged**